transport/http/err.go

```diff
@@ -0,0 +1,22 @@
+package http
+
+import (
+	"fmt"
+)
+
+// Err represents an Error occurred in the Client transport level.
+type Err struct {
```

Comment:
This should be of the form `SomethingError`, e.g. `TransportError`? See https://talks.golang.org/2014/names.slide#15